### PR TITLE
Sn models user post

### PR DESCRIPTION
**Pull request by DevXCoder47**
Models User and Post implemented

### CODE REVIEW

File: SocialNetwork/SocialNetwork.Core/Models/Post.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace SocialNetwork.Core.Models
+{
+    public class Post
+    {
+        public int Id { get; set; }
+        public string? Description { get; set; }
+        public object? Media { get; set; }
```
Comment (DmytroProg) [inline SocialNetwork/SocialNetwork.Core/Models/Post.cs]: object isn't the best data type for images. Better to use string and store a link to the resource that contains media content

File: SocialNetwork/SocialNetwork.Core/Models/Post.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+        public long LikesCount { get; set; }
+        public long DislikesCount { get; set; }
+        public long RepostsCount { get; set; }
```
Comment (DmytroProg) [inline SocialNetwork/SocialNetwork.Core/Models/Post.cs]: Is it written in the task that we need to work with reposts?

File: SocialNetwork/SocialNetwork.Core/Models/Post.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿namespace SocialNetwork.Core.Models
+{
+    public class Post
+    {
+        public int Id { get; set; }
+        public string? Description { get; set; }
+        public object? Media { get; set; }
+        public long LikesCount { get; set; }
+        public long DislikesCount { get; set; }
```
Comment (DmytroProg) [inline SocialNetwork/SocialNetwork.Core/Models/Post.cs]: In the task it says to stora Likes. We don't want user to add Dislikes so we don't need to store them

File: SocialNetwork/SocialNetwork.Core/Models/User.cs [hunk 4]
```diff
@@ -3,4 +3,13 @@
 public class User
 {
     public int Id { get; set; }
+    public string Nickname { get; set; }
+    public string? Password { get; set; }
```
Comment (DmytroProg) [inline SocialNetwork/SocialNetwork.Core/Models/User.cs]: In what scenario password can be null?

File: SocialNetwork/SocialNetwork.Core/Models/User.cs [hunk 5]
```diff
@@ -3,4 +3,13 @@
...
+    public string? PhoneNumber { get; set; }
+    public bool IsLoggedIn { get; set; }
+    public string? UserDescription { get; set; }
+    public string? UserIconFileName { get; set; }
+    public List<Post>? Posts { get; set; }
+    public DateTime WasCreated { get; set; }
```
Comment (DmytroProg) [inline SocialNetwork/SocialNetwork.Core/Models/User.cs]: Rename to 'CreatedAt'

File: SocialNetwork/SocialNetwork.Core/Models/User.cs [hunk 6]
```diff
@@ -3,4 +3,13 @@
 public class User
 {
     public int Id { get; set; }
+    public string Nickname { get; set; }
+    public string? Password { get; set; }
+    public string? PhoneNumber { get; set; }
+    public bool IsLoggedIn { get; set; }
+    public string? UserDescription { get; set; }
+    public string? UserIconFileName { get; set; }
+    public List<Post>? Posts { get; set; }
```
Comment (DmytroProg) [inline SocialNetwork/SocialNetwork.Core/Models/User.cs]: It is better to use ICollection<Post> instead

**Review by DmytroProg — CHANGES_REQUESTED** [review]
review comments

File: SocialNetwork/SocialNetwork.Core/Models/User.cs (hunk 4 above)
Comment (DevXCoder47) [reply SocialNetwork/SocialNetwork.Core/Models/User.cs]: Password may not be essential for creating an account

**Pull request closed without merging**